jumpcloud-ADMU/Powershell/Public/Start-Reversion.ps1

```diff
@@ -0,0 +1,391 @@
+        1. Looking up the account SID in the Windows registry ProfileList
+        2. Identifying the .ADMU profile path
+        3. Restoring original NTUSER.DAT and UsrClass.dat files from backups
+        4. Preserving migrated files with _migrated suffix for rollback purposes
+
+    .PARAMETER UserSID
+        The Windows Security Identifier (SID) of the user account to revert.
+
+    .PARAMETER TargetProfileImagePath
+        The actual profile path to revert. If not specified, will use the path from the registry.
+        This path will be validated to ensure it exists and is associated with the UserSID.
+
+    .PARAMETER DryRun
+        Shows what actions would be performed without actually executing them.
+
+    .PARAMETER Force
+        Bypasses confirmation prompts and forces the revert operation.
+
+    .EXAMPLE
+        Revert-Migration -UserSID "S-1-5-21-123456789-1234567890-123456789-1001"
+        Reverts the migration for the specified user SID using the registry profile path.
+
+    .EXAMPLE
+        Revert-Migration -UserSID "S-1-5-21-123456789-1234567890-123456789-1001" -TargetProfileImagePath "C:\Users\john.doe"
+        Reverts the migration using a specific target profile path instead of the registry value.
+
+    .EXAMPLE
+        Revert-Migration -UserSID "S-1-5-21-123456789-1234567890-123456789-1001" -DryRun
+        Shows what would be reverted without making actual changes.
+
+    .OUTPUTS
+        [PSCustomObject] Returns revert operation results with success status and details.
+    #>
+    [CmdletBinding(SupportsShouldProcess)]
+    param(
+        [Parameter(Mandatory = $true, Position = 0)]
+        [ValidatePattern("^S-\d-\d+-(\d+-){1,14}\d+$")]
+        [string]$UserSID,
+
+        [Parameter(Mandatory = $false, Position = 1)]
+        [ValidateScript({
+                if (Test-Path $_ -PathType Container) { $true }
+                else { throw "Target profile path does not exist: $_" }
+            })]
+        [string]$TargetProfileImagePath,
+
+        [Parameter(Mandatory = $false)]
+        [switch]$DryRun,
+
+        [Parameter(Mandatory = $false)]
+        [switch]$Force
+    )
+
+    Begin {
+        Write-ToLog -Message "Begin Revert Migration" -MigrationStep -Level Info -Step "Revert-Migration"
+
+        # Initialize result object
+        $revertResult = [PSCustomObject]@{
+            Success             = $false
+            UserSID             = $UserSID
+            RegistryProfilePath = $null
+            TargetProfilePath   = $TargetProfileImagePath
+            ActualProfilePath   = $null
+            FilesReverted       = @()
+            Errors              = @()
+            StartTime           = Get-Date
+            EndTime             = $null
+            WhatIfMode          = $DryRun.IsPresent
+            RegistryUpdated     = $false
+        }
+
+        # Regex pattern to identify .ADMU profile paths
+        $admuPathPattern = '\.ADMU$'
```

Comment:
Has this changed in start-migration yet?

Reply:
Yes it has https://github.com/TheJumpCloud/jumpcloud-ADMU/blob/7a90d808159b3a55cb899f7d4de3ac21a5be54f5/jumpcloud-ADMU/Powershell/Public/Start-Migration.ps1#L1257